src/main/java/org/apache/commons/compress/archivers/cpio/CpioArchiveInputStream.java

```diff
@@ -362,7 +362,11 @@ public int read(final byte[] b, final int off, final int len) throws IOException
 
     private long readAsciiLong(final int length, final int radix) throws IOException {
         final byte[] tmpBuffer = readRange(length);
-        return Long.parseLong(ArchiveUtils.toAsciiString(tmpBuffer), radix);
+        try {
```

Comment:
Hi @yakovsh 
It turns out, we do this kind of Long parse-and-rethrow-IOException in at least three other places. I'd think this should be refactored. Also, in total, there are ten call sites for Long parsing. Shouldn't all these be addressed uniformly in this PR? There are also Integer.parseXXX() calls that likely need the same treatment. WDYT?

Reply:
Hello @yakovsh 
It can be moved, sure. I would not use ExactMath since that's supposed to be a wrapper for Math exact APIs. If the refactored new API can be package-private, then please do that, if not, we'll have to live with a larger public API footprint. TY!